src/dsnote_app.cpp

```diff
@@ -1022,35 +1023,28 @@ void dsnote_app::handle_stt_text_decoded(QString text, const QString &lang,
         case text_destination_t::active_window:
 #ifdef USE_DESKTOP
             try {
+                // Make sure keyboard exists!
+                m_fake_keyboard.emplace();
+
                 // If this request was started as paste-to-active-window, copy
                 // to clipboard and paste via Ctrl+V
                 bool paste_mode =
                     settings::instance()->text_to_window_method() ==
                     settings::text_to_window_method_t::TextToWindowMethodCtrlV;
                 if (paste_mode) {
-                    // preserve current clipboard text, copy recognized text,
-                    // paste via Ctrl+V, then restore previous clipboard content
-                    // after a short delay
-                    auto *clip = QGuiApplication::clipboard();
-                    const QString prev_clip_text = clip->text();
-                    // set recognized text to clipboard
-                    clip->setText(text);
+                    // Copy text to clipboard and paste via Ctrl+V. Then restore
+                    // it
+                    auto prev_clip_text =
+                        m_fake_keyboard->copy_to_clipboard(text);
 
-                    m_fake_keyboard.emplace();
-                    m_fake_keyboard->send_ctrl_v();
+                    QTimer::singleShot(0, [this, prev_clip_text]{
+                        m_fake_keyboard->send_ctrl_v();
```

Comment:
```suggestion
                        if (!m_fake_keyboard) return;
                        m_fake_keyboard->send_ctrl_v();
```

Theoretically, m_fake_keyboard (an optional) might be empty.